File: selenium-framework/src/test/java/com/selenium/testng/enums/Menu.java
package com.selenium.testng.enums;

public enum Menu {

	ADMIN("Admin"), 
	PIM("PIM"), 
	LEAVE("Leave"), 
	TIME("Time"), 
	PERFORMANCE("Performance"), 
	DIRECTORY("Directory"),
	MAINTENANCE("Maintenance"), 
	CLAIM("Claim"), 
	BUZZ("Buzz");

	private final String value;

	// setting up enum constructor
	Menu(String value) {
		this.value = value;
	}

	public String getValue() {
		return value;
	}
}

File: src/framework/web/enums/menu.ts
export class Menu {
  static readonly ADMIN = new Menu('Admin');
  static readonly DASHBOARD = new Menu('Dashboard');
  static readonly LEAVE = new Menu('Leave');
  static readonly PIM = new Menu('PIM');
  static readonly TIME = new Menu('Time');
  static readonly RECRUITMENT = new Menu('Recruitment');
  static readonly MY_INFO = new Menu('My Info');

  private constructor(private readonly value: string) {}

  getValue(): string {
    return this.value;
  }
}
